subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/DependencyHandler.java

```diff
@@ -444,13 +446,22 @@
     void registerTransform(Action<? super VariantTransform> registrationAction);
 
     /**
-     * Registers an artifact transform.
+     * Registers an artifact transform with a parameter.
      *
-     * @see org.gradle.api.artifacts.transform.ArtifactTransform
+     * @see org.gradle.api.artifacts.transform.ArtifactTransformAction
      * @since 5.2
      */
     @Incubating
-    <T> void registerTransform(Class<T> configurationType, Action<? super ArtifactTransformSpec<T>> registrationAction);
+    <T> void registerTransform(Class<T> parameterType, Action<? super ArtifactTransformParameterSpec<T>> registrationAction);
```

Comment:
`ParametrizedArtifactTransformSpec`? `ArtifactTransformSpecWithParameters`?

Reply:
`ParameterizedArtifactTransformSpec` it is. I added the `e` (`terized` instead of `trized`) since we have been using this spelling somewhere else in the code base already.